IronFoundry.Warden/Containers/ContainerDirectory.cs

```diff
@@ -72,86 +96,45 @@ public override string ToString()
+            fileSystem.AddDirectoryAccess(containerPath, FileAccess.ReadWrite, user.UserName);
 
-            DirectoryInfo containerBaseInfo = dirInfo.Item1;
-            DirectorySecurity security = containerBaseInfo.GetAccessControl();
-            security.AddAccessRule(accessRule);
-
-            string containerDirectory = dirInfo.Item2;
-            return Directory.CreateDirectory(containerDirectory, security);
+            return new DirectoryInfo(containerPath);
         }
 
-        private static DirectoryInfo FindContainerDirectory(string containerBaseDirectory, ContainerHandle handle)
+        private static DirectoryInfo FindContainerDirectory(string containerPath)
         {
-            var dirInfo = GetContainerDirectoryInfo(containerBaseDirectory, handle);
-            if (Directory.Exists(dirInfo.Item2))
+            if (Directory.Exists(containerPath))
             {
-                return new DirectoryInfo(dirInfo.Item2);
+                return new DirectoryInfo(containerPath);
             }
             else
             {
-                throw new WardenException("Directory '{0}' does not exist!", dirInfo.Item2);
+                throw new WardenException("Directory '{0}' does not exist!", containerPath);
             }
         }
 
-        private static Tuple<DirectoryInfo, string> GetContainerDirectoryInfo(string containerBaseDirectory, ContainerHandle handle)
-        {
-            string containerDirectory = Path.Combine(containerBaseDirectory, handle);
-
-            return new Tuple<DirectoryInfo, string>(new DirectoryInfo(containerBaseDirectory), containerDirectory);
-        }
-
-        private void AddAccessRuleTo(FileSystemAccessRule accessRule, string path)
+        /// <summary>
+        /// Return the default access to use for new directories.  
+        /// </summary>
+        private IEnumerable<UserAccess> GetDefaultDirectoryAccess()
         {
-            var pathInfo = new DirectoryInfo(path);
-            if (pathInfo.Exists)
+            return new[]
             {
-                var pathSecurity = pathInfo.GetAccessControl();
-                pathSecurity.AddAccessRule(accessRule);
-
-                ReplaceAllChildPermissions(pathInfo, pathSecurity);
-            }
-            else
-            {
-                DirectoryInfo parentInfo = pathInfo.Parent;
-                if (parentInfo.Exists)
-                {
-                    var pathSecurity = parentInfo.GetAccessControl();
-                    pathSecurity.AddAccessRule(accessRule);
-
-                    Directory.CreateDirectory(pathInfo.FullName, pathSecurity);
-
-                    ReplaceAllChildPermissions(pathInfo, pathSecurity);
-                }
-            }
+                new UserAccess {UserName = GetBuiltInAdminGroupName(), Access = FileAccess.ReadWrite},
+                new UserAccess {UserName = user.UserName, Access = FileAccess.ReadWrite},
```

Comment:
The default access should not contain the container user.  We add it explicitly after the directory is created (above).